src/linear_algebra/matrix/schur_complement.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2022 Alexander Bentkamp. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alexander Bentkamp, Jeremy Avigad, Johan Commelin
+-/
+import linear_algebra.matrix.symmetric
+import linear_algebra.matrix.nonsingular_inverse
+import linear_algebra.matrix.pos_def
+
+/-! # Schur complement
+
+This file proves properties of the Schur complement `D - C A⁻¹ B` of a block matrix `[A B; C D]`.
+
+## Main result
+
+ * `matrix.schur_complement_pos_semidef_iff` :
+  If a matrix `A` is positive definite, then
+  `[A B; Bᴴ D]` is postive semidefinite if and only if `D - Bᴴ A⁻¹ B` is postive semidefinite.
+
+-/
+
+namespace matrix
+
+open_locale matrix
+variables {n : Type*} {R : Type*} [comm_ring R] [star_ring R]
+
+variables (A B C D : matrix n n R) (x y u v: n → R)
+
+localized "infix ` ⊕ᵥ `:65 := sum.elim" in matrix
+
+lemma star_sum_elim : star (x ⊕ᵥ y)  = (star x ⊕ᵥ star y) :=
+by { ext x, cases x; simp }
+
+lemma schur_complement_eq [fintype n] [decidable_eq n] {A : matrix n n R} [invertible A]
+  (hA : A.is_hermitian) :
+vec_mul (star (x ⊕ᵥ y)) (from_blocks A B Bᴴ D) ⬝ᵥ (x ⊕ᵥ y) =
+  vec_mul (star (x + (A⁻¹ ⬝ B).mul_vec y)) A ⬝ᵥ (x + (A⁻¹ ⬝ B).mul_vec y) +
+    vec_mul (star y) (D - Bᴴ ⬝ A⁻¹ ⬝ B) ⬝ᵥ y :=
+begin
+  simp [star_sum_elim, from_blocks_mul_vec, vec_mul_from_blocks, add_vec_mul, dot_product_mul_vec,
+    vec_mul_sub, matrix.mul_assoc, vec_mul_mul_vec, hA.eq, conj_transpose_nonsing_inv,
+    star_mul_vec],
+  abel
+end
+
+end matrix
+
+namespace matrix
+
+open_locale matrix
+variables {n : Type*} [fintype n] {R : Type*} [ordered_comm_ring R] [star_ring R]
+
+variables {A : matrix n n R} (B C D : matrix n n R) (x y u v: n → R)
+
+lemma schur_complement_is_hermitian_iff [decidable_eq n] (hA : A.is_hermitian) :
+  (from_blocks A B Bᴴ D).is_hermitian ↔ (D - Bᴴ ⬝ A⁻¹ ⬝ B).is_hermitian :=
```

Comment:
This is backwards, `A ↔ B` should be called `A_iff_B` or `A_iff` not `B_iff`:
```suggestion
lemma is_hermitian.from_blocks₁₁ [decidable_eq n] (hA : A.is_hermitian) :
  (from_blocks A B Bᴴ D).is_hermitian ↔ (D - Bᴴ ⬝ A⁻¹ ⬝ B).is_hermitian :=
```
or
```suggestion
lemma is_hermitian.schur_complement_iff [decidable_eq n] (hA : A.is_hermitian) :
  (D - Bᴴ ⬝ A⁻¹ ⬝ B).is_hermitian ↔ (from_blocks A B Bᴴ D).is_hermitian :=
```
I prefer the first one because it matches [`matrix.det_from_blocks₁₁`](https://leanprover-community.github.io/mathlib_docs/linear_algebra/matrix/nonsingular_inverse.html#matrix.det_from_blocks%E2%82%81%E2%82%81), and indicates that you should add the version with `A` and `D` swapped too.

Reply:
Ok, I have added all the variants.